### PR TITLE
docs(useSnapshot): Clarify parent/child useSnapshot usage.

### PR DESCRIPTION
**Pull request by stephenh**
## Related Issues or Discussions

Potentially clarifies behavior discussion from #673.

## Summary

Documents the `snap.books.map(book, i)` approach that @dai-shi mentioned is current best practice for passing stores to children.

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

File: docs/api/basic/useSnapshot.mdx
Comment (dai-shi) [inline docs/api/basic/useSnapshot.mdx:20]: we want to emphasize, when reading in a function (like `handleClick`), we should read it from the proxy state.

btw, I don't think we (should) use the term `store` in docs.

File: docs/api/basic/useSnapshot.mdx
Comment (dai-shi) [inline docs/api/basic/useSnapshot.mdx:71]: btw, we could pass both snap and state.

File: docs/api/basic/useSnapshot.mdx
Comment (dai-shi) [inline docs/api/basic/useSnapshot.mdx:67]: maybe, we should have a special section about memo and deps. it's very important but missing in docs, I suppose (kind of my bad.)

**Review by dai-shi — COMMENTED** [review]
I appreciate for your work! It's very very helpful.
I'd like to adjust something before merging.

File: docs/api/basic/useSnapshot.mdx
Comment (stephenh) [reply docs/api/basic/useSnapshot.mdx:81]: Sounds good, I pushed an update with a "passing both" example.

File: docs/api/basic/useSnapshot.mdx
Comment (stephenh) [reply docs/api/basic/useSnapshot.mdx]: Good point, I reworded + gave example of the callback using the proxy state.

> use the term store in docs.

Wdyt of the term state?

My hesitation about the term proxy is, ironically, the same hesitation you had wrt getters: that it seems like an advanced term/concept most users probably won't know about / shouldn't have to know about, as ideally/imo that Valtio uses proxies is just an implementation detail.

Vs. "store", which I think is generally widely used in state management circles for "the store of your state". So it's a more "logical"/concept based term than proxy's physical term.

...I would even pitch that in some Valtio 2.x, maybe `proxy` is renamed to `store`. :-D

("state" I think is too ambigious...do you mean my store state, or my snapshot state? Both the proxy and snapshot have state, it's just that one is mutable state and other immutable state. Hence like the store (mutable) + snapshot (immutable) pair.) 

That said, don't want that terminology suggestion to hold up the PR, so I've replaced all the "store"s with "proxy", unless you find one I've missed.

File: docs/api/basic/useSnapshot.mdx
Comment (stephenh) [reply docs/api/basic/useSnapshot.mdx:77]: Yeah, I agree; fine to be a separate PR?

File: docs/api/basic/useSnapshot.mdx
Comment (dai-shi) [reply docs/api/basic/useSnapshot.mdx]: I use "state" for valtio proxy objects and "snap" or "snapshot" for valtio snapshot objects.
I know "state" can be confusing and too general, though. We can use "proxy state" to be a little more explicit.

File: docs/api/basic/useSnapshot.mdx
Comment (dai-shi) [reply docs/api/basic/useSnapshot.mdx:77]: okay.

File: docs/api/basic/useSnapshot.mdx
Comment (dai-shi) [inline docs/api/basic/useSnapshot.mdx:117]: Do you mean between passing snap and passing proxy state?
Both are render optimized, but the latter with React.memo would perform better with large list. Yeah, but without memo, it's basically no difference (the former might be a bit efficient?). Using memo is kind of advanced, because we can't pass snapshot objects to memoed components. So, maybe it's a different story.

**Review by dai-shi — APPROVED** [review]
looks good to me. can merge if you are fine.

**Pull request merged**
